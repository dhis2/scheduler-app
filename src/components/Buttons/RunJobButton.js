import React from 'react'
import { func, string, bool } from 'prop-types'
import { connect } from 'react-redux'
import { Button } from '@dhis2/ui-core'
import * as rootSelectors from '../../rootSelectors'
import { selectors } from '../../data/jobs'
import { actions } from '../../data/modal'
import { modalTypes } from '../Modal'

export const DumbRunJobButton = ({ id, showModal, isFetching }) => {
    return (
        <Button
            primary
            name={`run-job-${id}`}
            disabled={isFetching}
            onClick={() =>
                showModal({ type: modalTypes.RUN_JOB, props: { id } })
            }
        >
            Run manually
        </Button>
    )
}

DumbRunJobButton.propTypes = {
    id: string.isRequired,
    isFetching: bool.isRequired,
    showModal: func.isRequired,
}

const mapStateToProps = state => {
    /* istanbul ignore next */
    const jobs = rootSelectors.getJobs(state)

    /* istanbul ignore next */
    return {
        isFetching: selectors.getIsFetching(jobs),
    }
}

const mapDispatchToProps = {
    showModal: actions.showModal,
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DumbRunJobButton)
