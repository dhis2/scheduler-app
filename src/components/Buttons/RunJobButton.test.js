import React from 'react'
import { shallow, mount } from 'enzyme'
import { DumbRunJobButton as RunJobButton } from './RunJobButton'

describe('<RunJobButton>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <RunJobButton id="1" isFetching={false} showModal={() => {}} />
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('calls showModal correctly when clicked', () => {
        const showModal = jest.fn()
        const wrapper = mount(
            <RunJobButton id="1" isFetching={false} showModal={showModal} />
        )

        wrapper.find('button').simulate('click')

        expect(showModal.mock.calls[0][0]).toMatchSnapshot()
    })

    it('disables itself whilst jobs are fetching', () => {
        const showModal = () => {}
        const wrapper = mount(
            <RunJobButton id="1" isFetching={true} showModal={showModal} />
        )

        const button = wrapper.find('button')

        expect(button.props().disabled).toBe(true)
    })
})
