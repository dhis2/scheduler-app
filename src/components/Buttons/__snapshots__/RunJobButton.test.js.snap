// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RunJobButton> calls showModal correctly when clicked 1`] = `
Object {
  "props": Object {
    "id": "1",
  },
  "type": "RUN_JOB",
}
`;

exports[`<RunJobButton> renders correctly 1`] = `
<Button
  disabled={false}
  name="run-job-1"
  onClick={[Function]}
  primary={true}
  type="button"
>
  Run
</Button>
`;
